wagtail/contrib/modeladmin/menus.py

```diff
@@ -26,6 +26,9 @@ def __init__(self, model_admin, order):
         )
 
     def is_shown(self, request):
+        if self.model_admin.exclude_from_menu is True:
+            return False
```

Comment:
I think it would be cleaner to handle this in [`WagtailRegisterable.register_with_wagtail`](https://github.com/wagtail/wagtail/blob/1f43d8ef51e455b92e42447fdc190d5ec83ec53c/wagtail/contrib/modeladmin/options.py#L61-L69), so that if both options are false, we never register the menu item at all - rather than register a menu item that always returns False from is_shown:

```python
        if self.add_to_settings_menu:
            menu_hook = "register_settings_menu_item"
        elif self.add_to_admin_menu:
            menu_hook = "register_admin_menu_item"
        else:
            menu_hook = None

        if menu_hook:
            @hooks.register(menu_hook)
            def register_admin_menu_item():
                return self.get_menu_item()
```

Reply:
@gasman Thanks, to clarify, this would mean `add_to_admin_menu` would be a property of `WagtailRegisterable` rather than `ModelAdmin` ?

Reply:
Yep, that's right.